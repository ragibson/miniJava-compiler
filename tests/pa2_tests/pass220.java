//  PA2 pass references
class T {
    void foo() {
	foo(4);
        this.foo(5);
        return p[3] + a.b;
    }
}
