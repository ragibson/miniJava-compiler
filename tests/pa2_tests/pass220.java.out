********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "T" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "foo" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [3]
  .   .   . CallStmt
  .   .   .   IdRef
  .   .   .     "foo" Identifier
  .   .   .   ExprList [1]
  .   .   .   . LiteralExpr
  .   .   .   .   "4" IntLiteral
  .   .   . CallStmt
  .   .   .   QualRef
  .   .   .     "foo" Identifier
  .   .   .     ThisRef
  .   .   .   ExprList [1]
  .   .   .   . LiteralExpr
  .   .   .   .   "5" IntLiteral
  .   .   . ReturnStmt
  .   .   .   BinaryExpr
  .   .   .     "+" Operator
  .   .   .       RefExpr
  .   .   .         IxRef
  .   .   .           IdRef
  .   .   .             "p" Identifier
  .   .   .           LiteralExpr
  .   .   .             "3" IntLiteral
  .   .   .       RefExpr
  .   .   .         QualRef
  .   .   .           "b" Identifier
  .   .   .           IdRef
  .   .   .             "a" Identifier
=============================================
